integration-test/helpers/objectSchema.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Returns the schema of a given JavaScript Object in the given Page. Schema is
+ * similar to the output of Object.getOwnPropertyDescriptors.
+ * Notes:
+ *  - Does not traverse up the target Object's __proto__ chain. If that is
+ *    desired, call `getObjectSchema` again for the Object's prototype.
+ *  - Functions are represented as a string "function(N)", where N is the number
+ *    of arguments that the function accepts (aka the function's length).
+ *  - Object keys are sorted alphabetically.
+ *  - Object values are handled recursively.
+ * @param {Page} page
+ *   The Puppeteer page to inspect.
+ * @param {string} targetObjectName
+ *   The name of the Object to inspect. A string which when eval'd in the page
+ *   returns the target Object. For example, "window.someApi".
+ * @returns {Object}
+ *   The target Object's schema.
+ */
+function getObjectSchema (page, targetObjectName) {
+    return page.evaluate(function getObjectSchema (targetObjectName) {
+        const result = Object.create(null)
+        // eslint-disable-next-line no-eval
+        const targetObject = eval(targetObjectName)
+
+        const descriptors = Object.getOwnPropertyDescriptors(targetObject)
+        for (const propertyName of Object.keys(descriptors).sort()) {
+            const descriptor = descriptors[propertyName]
+            result[propertyName] = Object.create(null)
+            for (const key of Object.keys(descriptor).sort()) {
+                const value = descriptor[key]
+                if (typeof value === 'function') {
+                    result[propertyName][key] = 'function(' + value.length + ')'
+                } else if (typeof value === 'object' && value !== null) {
+                    result[propertyName][key] = getObjectSchema(value)
+                } else {
+                    result[propertyName][key] = JSON.stringify(value)
```

Comment:
FYI stringify will also fail if there is a circular reference (try `JSON.stringify(window)`)

Reply:
Ack, I think this is OK now.